packages/dashboard-core-plugins/src/panels/FileExplorerPanel.tsx

```diff
@@ -157,6 +159,31 @@ export class FileExplorerPanel extends React.Component<
     fileStorage.createDirectory(path).catch(FileExplorerPanel.handleError);
   }
 
+  async handleCopyFile(file: FileStorageItem): Promise<void> {
+    const { fileStorage } = this.props;
+    if (isDirectory(file)) {
+      log.error('Invalid item in handleCopyItem', file);
+      return;
+    }
+    let newName = FileUtils.getCopyFileName(file.filename);
+    const checkNewName = async (): Promise<boolean> => {
+      try {
+        await fileStorage.info(newName);
+        return true;
+      } catch (error) {
+        return false;
+      }
+    };
+    let newNameExists = await checkNewName();
+    while (newNameExists) {
+      newName = FileUtils.getCopyFileName(newName);
+      // await in loop is fine here, this isn't a parallel task
+      // eslint-disable-next-line no-await-in-loop
+      newNameExists = await checkNewName();
+    }
```

Comment:
You can simplify this a bit:
```suggestion    
    // await in loop is fine here, this isn't a parallel task
    // eslint-disable-next-line no-await-in-loop
    while (await checkNewName()) {
      newName = FileUtils.getCopyFileName(newName);
    }
```